onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp

```diff
@@ -65,7 +65,7 @@ struct OperatorRegistrationInformation
     gsl::span<const SupportedTensorDataTypes> supportedTensorDataTypes;
     DmGraphSupport DmGraphSupport;
 
-    std::vector<uint32_t> requiredConstantCpuInputs;
+    std::pair<std::array<const uint32_t, 4>, int> requiredConstantCpuInputs = {{}, 0};
```

Comment:
I hardcoded the limit of 4, since it's the largest size used currently. Other options would be bumping to C++20 (which has constexpr vector) or instantiating the data outside of `OperatorRegistrationInformation` and holding a std::span (C++20) or gsl::span to it.

Creating a vector of 4 ints for each operator seemed reasonable and is the simplest option implementation-wise. The span option would save a negligible amount of bytes (small amount in operators that don't use requiredConstantCpuInputs, since the pointer kept in the span is likely < 32 * 4 bits, but would waste more space in operators that use e.g. 4 requiredConstantCpuInputs, since we now have to keep a list of the 4 indexes and a pointer to the list)

Reply:
I'm not sure if initializer_list works here... It's only creating a temporary object and not instantiating the data statically anywhere. Following code:

```
constexpr auto a = ConstExprSpan{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20};
constexpr auto b = ConstExprSpan(std::initializer_list<int>{0, 1, 2});
```

Gives me:

```
<source>: In instantiation of 'constexpr ConstExprSpan<T>::ConstExprSpan(std::initializer_list<_Tp>) [with T = int]':

<source>:26:106:   required from here

<source>:10:80: warning: initializing 'ConstExprSpan<int>::m_begin' from 'std::initializer_list<int>::begin' does not extend the lifetime of the underlying array [-Winit-list-lifetime]

   10 |     constexpr ConstExprSpan(std::initializer_list<T> list) : m_begin(list.begin()), m_end(list.end()) {}

      |                                                                      ~~~~~~~~~~^~

<source>:26:106: error: 'ConstExprSpan<int>{((const int*)(&<anonymous>)), (((const int*)(&<anonymous>)) + 84)}' is not a constant expression

   26 | constexpr auto a = ConstExprSpan{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20};

      |                                                                                                          ^

<source>:27:69: error: 'ConstExprSpan<int>{((const int*)(&<anonymous>)), (((const int*)(&<anonymous>)) + 12)}' is not a constant expression

   27 | constexpr auto b = ConstExprSpan(std::initializer_list<int>{0, 1, 2});
```

in GCC